Makefile

```diff
@@ -172,9 +171,36 @@ docker-push-manifest: ## Push the fat manifest docker image.
 	## Minimum docker version 18.06.0 is required for creating and pushing manifest images.
 	docker manifest create --amend $(CONTROLLER_IMG):$(TAG) $(shell echo $(ALL_ARCH) | sed -e "s~[^ ]*~$(CONTROLLER_IMG)\-&:$(TAG)~g")
 	@for arch in $(ALL_ARCH); do docker manifest annotate --arch $${arch} ${CONTROLLER_IMG}:${TAG} ${CONTROLLER_IMG}-$${arch}:${TAG}; done
-	docker manifest push --purge ${CONTROLLER_IMG}:${TAG}
-	@echo "updating kustomize image patch file for manager resource"
-	sed -i'' -e 's@image: .*@image: '"${CONTROLLER_IMG}:$(TAG)"'@' ./config/default/manager_image_patch.yaml
+	MANIFEST_IMG=$(CONTROLLER_IMG) MANIFEST_TAG=$(TAG) $(MAKE) set-manifest-image
+
+.PHONY: set-manifest-image
+set-manifest-image:
+	$(info Updating kustomize image patch file for manager resource)
+	sed -i'' -e 's@image: .*@image: '"${MANIFEST_IMG}:$(MANIFEST_TAG)"'@' ./config/default/manager_image_patch.yaml
+
+## --------------------------------------
+## Release
+## --------------------------------------
+
+RELEASE_TAG := $(shell git tag --contains HEAD | tail -n 1)
+
+.PHONY: release
+release:  ## Builds and push container images using the latest git tag for the commit.
+	@if [ -z "${RELEASE_TAG}" ]; then echo "RELEASE_TAG is not set"; exit 1; fi
+	# Push the release image to the staging bucket first.
+	REGISTRY=gcr.io/k8s-staging-cluster-api TAG=$(RELEASE_TAG) \
+		$(MAKE) docker-build-all docker-push-all
+	# Set the manifest image to the production bucket.
+	REGISTRY=us.gcr.io/k8s-artifacts-prod/cluster-api TAG=$(RELEASE_TAG) \
+		set-manifest-image
+	# Generate release artifacts.
+	mkdir out/
+	kustomize build config/default > out/provider-components.yaml
```

Comment:
s/provider-components/cluster-api-components/

Reply:
Should we try to find the same name to release these artifacts across all repos?

Reply:
Maybe according to use...  So for capi it would be `cluster-api-components` or `capi-components`, For a bootstrap provider it could be `bootstrap-components`, for an infrastructure provider it could be `infrastructure-components`.

Reply:
Fair enough!

Reply:
Fixed, good feedback :)